app/src/main/java/com/xmartlabs/template/service/gson/GsonRequired.java

```diff
@@ -0,0 +1,13 @@
+package com.xmartlabs.template.service.gson;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+/** Annotation for fields that must be present and have a value when deserialized using Gson */
```

Comment:
I think "Annotation for fields" is redundant here. Maybe "Indicates that a field ..."